### ISSUE TITLE
Should fieldset legends default to the attribute name?

### ISSUE THREAD

**Issue by peteryates** [DFE-Digital/govuk_design_system_formbuilder#142]
The behaviour of legends and labels differs in that 
* labels will always be rendered, defaulting to the attribute name if nothing is supplied
* legends won't be rendered if text isn't supplied (via an arg or by localisation)

In every case [_except_ for standalone fieldsets](https://govuk-form-builder.netlify.app/building-blocks/fieldsets/) the fieldset (and the legend) will be associated with a attribute. Perhaps it would make more sense to use it rather than just omitting the legend.

### No label or legend

```slim
= form_with(model: widget) do |form|

  = form.govuk_text_field :name,
    caption: { text: %(Name caption), size: %(m) }

  = form.govuk_date_field :created_at,
    caption: { text: %(Created on caption), size: %(m) }
```
![Screenshot from 2020-06-04 15-02-42](https://user-images.githubusercontent.com/128088/83766719-a9002e80-a674-11ea-868b-f557c9184848.png)

The label's caption appears alone, the legend's isn't rendered at all

### With label and legend

```slim
= form_with(model: widget) do |form|

  = form.govuk_text_field :name,
    label: { tag: %(h3), size: %(l) },
    caption: { text: %(Name caption), size: %(m) }

  = form.govuk_date_field :created_at,
    legend: { text: %(Created on), size: %(l) },
    caption: { text: %(Created on caption), size: %(m) }
```
![Screenshot from 2020-06-04 15-03-00](https://user-images.githubusercontent.com/128088/83766741-af8ea600-a674-11ea-8edd-f711ead3c326.png)

Both the label and legend captions are visible

-----

Is the current behaviour unpredictable? Is it correct? Feedback welcome. Thanks @rjlynch for reporting this.

**Comment by peteryates** [comment]
Having put some thought into this, the answer is they *probably* should do, for consistency with labels, but it's a change that could have unpredictable side effects for people upgrading.

This will be pushed back until the next major version.

**Comment by peteryates** [comment]
This work is done and will be released as part of #175.
